transaction/tests/StronglyConnectedComponents_unittest.cpp

```diff
@@ -0,0 +1,286 @@
+    tid6 = new transaction_id(6);
+    tid7 = new transaction_id(7);
+    tid8 = new transaction_id(8);
+    tid9 = new transaction_id(9);
+    tid10 = new transaction_id(10);
+    tid11 = new transaction_id(11);
+    tid12 = new transaction_id(12);
+
+    nid1 = wait_for_graph->addNodeUnchecked(tid1);
+    nid2 = wait_for_graph->addNodeUnchecked(tid2);
+    nid3 = wait_for_graph->addNodeUnchecked(tid3);
+    nid4 = wait_for_graph->addNodeUnchecked(tid4);
+    nid5 = wait_for_graph->addNodeUnchecked(tid5);
+    nid6 = wait_for_graph->addNodeUnchecked(tid6);
+    nid7 = wait_for_graph->addNodeUnchecked(tid7);
+    nid8 = wait_for_graph->addNodeUnchecked(tid8);
+    nid9 = wait_for_graph->addNodeUnchecked(tid9);
+    nid10 = wait_for_graph->addNodeUnchecked(tid10);
+    nid11 = wait_for_graph->addNodeUnchecked(tid11);
+    nid12 = wait_for_graph->addNodeUnchecked(tid12);
+
+    wait_for_graph->addEdgeUnchecked(nid1, nid2);
+
+    wait_for_graph->addEdgeUnchecked(nid2, nid3);
+    wait_for_graph->addEdgeUnchecked(nid2, nid4);
+    wait_for_graph->addEdgeUnchecked(nid2, nid5);
+
+    wait_for_graph->addEdgeUnchecked(nid3, nid6);
+
+    wait_for_graph->addEdgeUnchecked(nid4, nid5);
+    wait_for_graph->addEdgeUnchecked(nid4, nid7);
+
+    wait_for_graph->addEdgeUnchecked(nid5, nid2);
+    wait_for_graph->addEdgeUnchecked(nid5, nid6);
+    wait_for_graph->addEdgeUnchecked(nid5, nid7);
+
+    wait_for_graph->addEdgeUnchecked(nid6, nid3);
+    wait_for_graph->addEdgeUnchecked(nid6, nid8);
+
+    wait_for_graph->addEdgeUnchecked(nid7, nid8);
+    wait_for_graph->addEdgeUnchecked(nid7, nid10);
+
+    wait_for_graph->addEdgeUnchecked(nid8, nid7);
+
+    wait_for_graph->addEdgeUnchecked(nid9, nid7);
+
+    wait_for_graph->addEdgeUnchecked(nid10, nid9);
+    wait_for_graph->addEdgeUnchecked(nid10, nid11);
+
+    wait_for_graph->addEdgeUnchecked(nid11, nid12);
+
+    wait_for_graph->addEdgeUnchecked(nid12, nid10);
+
+    scc = std::make_unique<StronglyConnectedComponents>(*wait_for_graph);
+  }
+
+  std::unique_ptr<DirectedGraph> wait_for_graph;
+  std::unique_ptr<StronglyConnectedComponents> scc;
+
+  transaction_id *tid1;
+  transaction_id *tid2;
+  transaction_id *tid3;
+  transaction_id *tid4;
+  transaction_id *tid5;
+  transaction_id *tid6;
+  transaction_id *tid7;
+  transaction_id *tid8;
+  transaction_id *tid9;
+  transaction_id *tid10;
+  transaction_id *tid11;
+  transaction_id *tid12;
+
+  NID nid1;
+  NID nid2;
+  NID nid3;
+  NID nid4;
+  NID nid5;
+  NID nid6;
+  NID nid7;
+  NID nid8;
+  NID nid9;
+  NID nid10;
+  NID nid11;
+  NID nid12;
+
+  std::uint64_t nid1_component;
+  std::uint64_t nid2_component;
+  std::uint64_t nid3_component;
+  std::uint64_t nid4_component;
+  std::uint64_t nid5_component;
+  std::uint64_t nid6_component;
+  std::uint64_t nid7_component;
+  std::uint64_t nid8_component;
+  std::uint64_t nid9_component;
+  std::uint64_t nid10_component;
+  std::uint64_t nid11_component;
+  std::uint64_t nid12_component;
```

Comment:
P.S. Vectors have a nice constructor feature: `vector<int> v({1,2,3,4,5,6})` which will make the code prettier.